### ISSUE TITLE
POST request missing `Content-Type` header for empty files

### ISSUE THREAD

**Issue by yuriyyakym** [tus/tus-js-client#597]
**Description**

When uploading empty files, they are created in single POST request. Regardless `uploadDataDuringCreation` setting, no PATCH request is sent.
The problem is that POST request does not have 'Content-Type' header set, whereas @tus/server expects this header to be `application/offset+octet-stream`, otherwise `onUploadFinish` is not called:
[@tus/server POST handler](https://github.com/tus/tus-node-server/blob/main/packages/server/src/handlers/PostHandler.ts#L100-L114)
[@tus/server header validation](https://github.com/tus/tus-node-server/blob/main/packages/server/src/validators/HeaderValidator.ts#L79-L84)

According to specification `Content-Type` is required for PATCH requests, but not for POST requests, so I am not sure where to report this issue.

Potential solutions:
- setting `Content-Type: application/offset+octet-stream` for empty files uploads on client side
- update @tus/server to call `onUploadFinish` regardless `Content-Type` header (If this is the way to go, curent issue should be closed, and corresponding one created for @tus/server repo).

**To Reproduce**

1. Configure any @tus/server with some `onUploadFinish` callback logging
2. Set [uploadDataDuringCreation](https://github.com/tus/tus-js-client/blob/main/docs/api.md#uploaddataduringcreation) to false (default)
3. Create an empty text file
4. Upload

**Expected behavior**

`onUploadFinish` is called in @tus/server for empty file uploads

**Setup details**

- Used tus-js-client version: 3.1.0
- Used tus server software: tus-node-server 1.0.0-beta.6

**Comment by Acconut** [comment]
> * update @tus/server to call `onUploadFinish` regardless `Content-Type` header (If this is the way to go, curent issue should be closed, and corresponding one created for @tus/server repo).

Yes, this should be the correct approach. If a POST request with `Upload-Length: 0` and without `Upload-Defer-Length: 1` is received, the server should consider the upload to be complete. If tus-node-server does not handle this properly, it should be fixed there. I don't think this is a bug in tus-js-client. /cc @Murderlon
